src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
         {
-            HttpResponseMessage responseMessage = null;
+            CurlResponseMessage responseMessage = null;
             SafeCurlHandle requestHandle = null;
+            GCHandle requestStateHandle = new GCHandle();
+            IntPtr headerPtr = IntPtr.Zero;
+            bool referenceTaken = false;
```

Comment:
Actually I copied this pattern from WinHttpHandler which does an AddRef before calling an async operation. (eg. see constructor of WinHttpRequestStream and its usage in WinHttpHandler.StartRequest). Is this AddRef needed or can the CurlSendIoctlCallback assume that the handle cannot get released since StartRequest is blocked on curl_easy_perform?